app/routes/observations.tsx

```diff
@@ -9,34 +9,39 @@ import { parseLatitude, parseLongitude } from "~/utils/parser";
 
 export async function loader({ request }: LoaderFunctionArgs) {
   const url = new URL(request.url);
-  const startLongitude = parseLongitude(url.searchParams.get("startLongitude"));
-  const startLatitude = parseLatitude(url.searchParams.get("startLatitude"));
-  const endLongitude = parseLongitude(url.searchParams.get("endLongitude"));
+  const startLatitude = parseLatitude(url.searchParams.get("startLongitude"));
+  const startLongitude = parseLongitude(url.searchParams.get("startLatitude"));
   const endLatitude = parseLatitude(url.searchParams.get("endLatitude"));
+  const endLongitude = parseLongitude(url.searchParams.get("endLangitude"));
+
+  let observations;
 
   if (
-    startLongitude !== null &&
-    startLatitude !== null &&
-    endLongitude !== null &&
-    endLatitude !== null
+    startLatitude === null ||
+    startLongitude === null ||
+    endLatitude === null ||
+    endLongitude === null
   ) {
-    const observations = await getObservations({
+    observations = await getObservations();
```

Comment:
It was intentional to leave the functionality of fetching all observations out of this endpoint. It's meant only for the admin to be able to fetch everything at once.

Reply:
Fixed